jupyter_server_proxy/handlers.py

```diff
@@ -142,32 +136,37 @@ def _get_context_path(self, port):
         """
         if self.proxy_base:
             return url_path_join(self.base_url, self.proxy_base)
-        if self.rewrite == '/':
-            return url_path_join(self.base_url, 'proxy', str(port))
-        if self.rewrite == '':
+        if self.absolute_url:
             return url_path_join(self.base_url, 'proxy', 'absolute', str(port))
-        raise ValueError('Unsupported rewrite: "{}"'.format(self.rewrite))
+        else:
+            return url_path_join(self.base_url, 'proxy', str(port))
 
-    def _build_proxy_request(self, port, proxied_path, body):
+    def get_client_uri(self, protocol, port, proxied_path):
         context_path = self._get_context_path(port)
-        if self.rewrite:
-            client_path = proxied_path
-        else:
+        if self.absolute_url:
             client_path = url_path_join(context_path, proxied_path)
+        else:
+            client_path = proxied_path
 
-        client_uri = '{uri}:{port}{path}'.format(
-            uri='http://localhost',
+        client_uri = '{protocol}://{host}:{port}{path}'.format(
+            protocol=protocol,
+            host='localhost',
             port=port,
             path=client_path
         )
         if self.request.query:
             client_uri += '?' + self.request.query
 
+        return client_uri
+
+    def _build_proxy_request(self, port, proxied_path, body):
+
         headers = self.proxy_request_headers()
 
+        client_uri = self.get_client_uri('http', port, proxied_path)
         # Some applications check X-Forwarded-Context and X-ProxyContextPath
         # headers to see if and where they are being proxied from.
-        if self.rewrite == '/':
+        if not self.absolute_url:
```

Comment:
@yuvipanda Since the `_build_proxy_request` method has been split you'll need either `context_path = self._get_context_path(port)` here, or have `self.get_client_uri` return a tuple `(client_uri, context_path)`

Reply:
Fixed!

Reply:
Fixed!